examples/nidcpower_source_dc_voltage/_helpers.py

```diff
@@ -231,3 +232,111 @@ def use_simulation_option(default: bool) -> Callable[[F], F]:
         is_flag=True,
         help="Use simulated instruments.",
     )
+
+
+def create_driver_session(
+    measurement_service: MeasurementService,
+    pin_names: Iterable[str],
+    instrument_type_module: Any,
+    service_options: ServiceOptions,
+    instrument_type_id: str,
+) -> Tuple[Any, List[session_management.SessionInformation]]:
+    """Create and reserve driver sessions."""
+    session_management_client = nims.session_management.Client(
+        grpc_channel=measurement_service.get_channel(
+            provided_interface=nims.session_management.GRPC_SERVICE_INTERFACE_NAME,
+            service_class=nims.session_management.GRPC_SERVICE_CLASS,
+        )
+    )
+
+    with contextlib.ExitStack() as stack:
+        reservation = stack.enter_context(
+            reserve_session(
+                session_management_client,
+                measurement_service.context.pin_map_context,
+                instrument_type_id,
+                timeout=60,
+                pin_names=pin_names,
+            )
+        )
+
+        if len(reservation.session_info) != 1:
+            measurement_service.context.abort(
+                grpc.StatusCode.INVALID_ARGUMENT,
+                f"Unsupported number of sessions: {len(reservation.session_info)}",
+            )
+
+        # Leave session open.
+        session = stack.enter_context(
+            create_session(
+                reservation.session_info[0],
+                instrument_type_module,
+                service_options,
+                measurement_service,
+            )
+        )
+
+    return (session, reservation.session_info)
+
+
+def create_session(
+    session_info: session_management.SessionInformation,
+    instrument_type_module,
+    service_options: Optional[ServiceOptions] = None,
+    measurement_service: Optional[MeasurementService] = None,
+) -> Any:
+    """Create driver session based on the instrument type and reserved session."""
+    session_kwargs = {}
+    # To support teststand_fixture functions since they dont have service options
```

Comment:
I am adding session-based simulation code to these functions.

Reply:
Included session based simulation code in helper methods.